### PR TITLE
Item inspector

### PR DESCRIPTION
**Pull request by FaceDeer**
I use this all the time while developing my own mods and it's awesome. But I've been doing a lot of work with craft items and tools that have metadata lately, and this didn't handle those. So I added a formspec for inspecting items in my inventory. Use "/inspect_item" to bring it up.

While I was in here I also made a few other changes that I hope are all reasonable "cleanup" style tweaks, if any of them are unwelcome let me know and I'll retract this pull request and offer one without them. To wit:

* I made the fsc dependency optional. I usually develop mods on a very bare-bones world without other mods installed so I'd rather not have to include fsc just for this, and since there's no on_receive_fields method to secure in the first place I'm not sure there's much benefit to be had anyway.

* Added the LGPLv2.1 license text explicitly to the repository.

* Also, wrote up a simple README.

* Optimized the PNGs, because why not.

### CODE REVIEW

**Comment by FaceDeer** [comment]
I just added a quick bugfix to this pull, inspector was crashing when you tried inspecting an undefined node.

File: description.txt [hunk 1]
```diff
@@ -1 +1 @@
-An in-game tool to inspect node parameters and metadata.
+An in-game tool to inspect node and item parameters and metadata.
```
Comment (SmallJoker) [inline description.txt]: You may delete `depends.txt` and `description.txt`.
`remove_detached_inventory` was added in 5.0.0-dev, thus 0.4.x compatibility won't be needed.

File: init.lua [hunk 2]
```diff
@@ -289,7 +306,81 @@ minetest.register_chatcommand("inspect", {
...
+	local meta = itemstack:get_meta()
+	local metatable = meta:to_table()
+	desc = desc .. "==== meta ====\n"
+	desc = desc .. indent(1, "meta.fields = " .. adjusted_dump(metatable.fields)) .. "\n"
+	
+	local itemdef = itemstack:get_definition()
+	-- combine itemdef table with its "superclass" table
+	local combined_fields = {}
+	local nodedef_fields = {}
+	for key, value in pairs(getmetatable(itemdef).__index) do combined_fields[key] = value end
```
Comment (SmallJoker) [inline init.lua]: No benefit of squashing this loop. Please split this line into multiple lines to make it more readable.

**Comment by SmallJoker** [comment]
Will merge after the comments are handled and the PR re-tested.

**Comment by FaceDeer** [comment]
There were a bunch of other similarly squashed ifs and fors, I expanded those too. I left [this group](https://github.com/minetest-mods/inspector/pull/13/commits/b3861156ed72be9538ef493c2d2863cffa0edd05#diff-d7d74284e2e39f6c06d199c8e2bedbd0R117) alone, though, it seemed more elegant in its current form.

**Review by SmallJoker — APPROVED** [review]
Works.

**Pull request merged**